File: 101-Symmetric Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if not root:
            return True
        
        def dfs(leftNode, rightNode):
            if not leftNode and not rightNode:
                return True
            elif not leftNode or not rightNode:
                return False
            elif leftNode.val != rightNode.val:
                return False
            else:
                return dfs(leftNode.left, rightNode.right) and dfs(leftNode.right, rightNode.left)
                
        return dfs(root.left, root.right)

File: JavaScript/101-Symmetric Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    if (!root) return true;
    let dfs = (leftNode, rightNode) =>{
        if (!leftNode && !rightNode) return true;
        else if (!leftNode || !rightNode) return false;
        else if (leftNode.val !== rightNode.val) return false;
        else{
            return dfs(leftNode.left, rightNode.right) && dfs(leftNode.right, rightNode.left);
        }
    }
    return dfs(root.left, root.right);
};